clients/destination.go

```diff
@@ -1,18 +1,37 @@
 package clients
 
 import (
+	"bufio"
 	"context"
 	"encoding/json"
 	"fmt"
+	"os"
+	"os/exec"
 
 	"github.com/cloudquery/plugin-sdk/internal/pb"
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
 )
 
 type DestinationClient struct {
-	pbClient pb.DestinationClient
+	pbClient       pb.DestinationClient
+	cmd            *exec.Cmd
+	logger         zerolog.Logger
+	conn           *grpc.ClientConn
+	grpcSocketName string
+	cmdWaitErr     error
+}
+
+type DestinationClientOption func(*DestinationClient)
+
+func WithDestinationLogger(logger zerolog.Logger) func(*DestinationClient) {
+	return func(c *DestinationClient) {
+		c.logger = logger
+	}
 }
 
 func NewDestinationClient(cc grpc.ClientConnInterface) *DestinationClient {
```

Comment:
Should this function maybe also accept the new options?

Reply:
On the fence here, maybe in the future (in anycase Im less worried about it because it is mainly used by the sdk). specifaclly the log has not impact if the server is already running

Reply:
Ok yeah, let's leave it 👍